ansible/configs/open-training-novelloshell/pull_repo.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Ensure secret directory exists in output_dir
+  file:
+    path: "{{ output_dir }}/secrets"
+    state: directory
+
+- fail:
+    msg: >-
+      You must define heat_templates_private_repo
+  when: heat_templates_private_repo is not defined
+
+- fail:
+    msg: >-
+      You must define heat_templates_private_key_content in
+      order to pull the templates from the private repository.
+  when: heat_templates_private_key_content is not defined
+
+- name: block for decode key
+  when: platform == "babylon"
+  block:
+
+  - name: Copy content of the SSH key file
+    copy:
+      src: "{{ heat_templates_private_key_content  }}"
+      dest: "{{ output_dir }}/secrets/heat_templates_base64_private_key"
+      mode: 0400
+
+  - name: decode content
+    shell: >-
+      cat "{{ output_dir }}/secrets/heat_templates_base64_private_key"
+      | base64 --decode > "{{ output_dir }}/secrets/heat_templates_private_key"
+
+  - name: decode content
+    shell: chmod 0400 "{{ output_dir }}/secrets/heat_templates_private_key"
+
+- name: Copy content of the SSH key file
+  copy:
+    content: "{{ heat_templates_private_key_content  }}"
+    dest: "{{ output_dir }}/secrets/heat_templates_private_key"
+    mode: 0400
+  when: platform != "babylon"
+
+- name: Display the private repository and tag
+  ansible.builtin.debug:
+    msg:
+    - "Repository {{ heat_templates_private_repo }}"
+    - "Tag {{ heat_templates_private_repo_tag | default('master') }}"
+
+- name: Download the templates from the private repository
+  git:
+    repo: "{{ heat_templates_private_repo }}"
+    version: "{{ heat_templates_private_repo_tag | default('master') }}"
+    accept_hostkey: true
+    depth: 1
+    dest: "{{ output_dir }}/imported-templates"
+    key_file: "{{ output_dir }}/secrets/heat_templates_private_key"
```

Comment:
would it make sense to provide the template content upfront in agnosticv as a variable instead of pulling from a repo?